### PR TITLE
kernels: (streamer-matmul) use gemmx

### PR DESCRIPTION
**Pull request by jorendumoulin**
this pr replaces the streamer-gemm with the new gemmx accelerator for the kernels:

- dynamic tests are removed as they are not yet supported by the stream lowering
- half_tiled matmul is removed as the transform_matmul also covers this case

### CODE REVIEW

File: compiler/accelerators/snax_gemmx.py [hunk 1]
```diff
@@ -150,4 +151,54 @@ def _generate_setup_vals(
...
+            csr1 = c0
+            csr2 = c0
+
+            # get zero points for gemm
+            assert isinstance(qmac.zp_lhs, BlockArgument)
+            zp_a = generic_op.inputs[qmac.zp_lhs.index]
+            assert isinstance(qmac.zp_rhs, BlockArgument)
+            zp_b = generic_op.inputs[qmac.zp_rhs.index]
+
+            # bitwise and with 8b'11111111 to don't fuck up bitpacking in case of negative values
```
Comment (JosseVanDelm) [inline compiler/accelerators/snax_gemmx.py:182]:
```suggestion
            # bitwise and with 8b'11111111 to prevent extraneous sign extension
```

Comment (JosseVanDelm) [reply compiler/accelerators/snax_gemmx.py:182]: I don't understand why this is necessary actually. You can perform explicit unsigned extension of 8-bit values with extui?

File: compiler/transforms/schedule_memref_linalg.py [hunk 2]
```diff
@@ -141,6 +142,25 @@ def match_and_rewrite(self, op: memref_stream.GenericOp, rewriter: PatternRewrit
...
+                ]
+                template_bounds = (None, None, None, 8, 8, 8)
+            else:
+                # rescale function of gemmx
+                M, K, m, k = (AffineDimExpr(i) for i in range(4))
+                template = [
+                    AffineMap(4, 0, (M * 8 + m, K * 8 + k)),
+                    AffineMap(4, 0, (M * 8 + m, K * 8 + k)),
+                ]
+                template_bounds = (None, None, 8, 8)
```
Comment (JosseVanDelm) [inline compiler/transforms/schedule_memref_linalg.py:163]: Could you already write them to be movable?

File: compiler/transforms/stream_snaxify.py [hunk 3]
```diff
@@ -155,13 +156,52 @@ def generate_one_list(n: int, i: int):
...
+                        upper_bounds=snax_stride_patterns[3].upper_bounds,
+                        temporal_strides=[0] * 3,
+                        spatial_strides=[0, 4, 32],
+                    ),
+                )
+
+                # point C to zero allocated row in TCDM.
+                new_inputs.append(
+                    arith.Constant.from_int_and_width(0x1000_0040, builtin.IndexType())
+                )
```
Comment (JosseVanDelm) [inline compiler/transforms/stream_snaxify.py:199]: I'm wondering whether we should ask for hardware supported zero memory

**Review by JosseVanDelm — COMMENTED** [review]
Is dynamic shape support for streamer lowerings planned/in the works? I agree with moving forward, but this feels (maybe wrongly) like a reduction of our test surface area?

File: compiler/accelerators/snax_gemmx.py (hunk 1 above)
Comment (jorendumoulin) [reply compiler/accelerators/snax_gemmx.py:182]: The problem is that the values are in 32 bit format, after default linalg lowerings, so extui is not possible

File: compiler/transforms/schedule_memref_linalg.py (hunk 2 above)
Comment (jorendumoulin) [reply compiler/transforms/schedule_memref_linalg.py:163]: What does this mean?

File: compiler/transforms/stream_snaxify.py (hunk 3 above)
Comment (jorendumoulin) [reply compiler/transforms/stream_snaxify.py:199]: New streamers can be masked such that they output zero values, so we could use that.

**Comment by jorendumoulin** [comment]
> Is dynamic shape support for streamer lowerings planned/in the works? I agree with moving forward, but this feels (maybe wrongly) like a reduction of our test surface area?

Indeed, this removes a test, but for a lowering I feel we don't want to support. I don't immediately plan on dynamic lowerings for streamers, but can do it when necessary

File: compiler/transforms/schedule_memref_linalg.py (hunk 2 above)
Comment (JosseVanDelm) [reply compiler/transforms/schedule_memref_linalg.py:163]: oeps

File: compiler/accelerators/snax_gemmx.py (hunk 1 above)
Comment (JosseVanDelm) [reply compiler/accelerators/snax_gemmx.py:182]: Right. Okay :) . Still would like the comment to be a bit more descriptive :)

**Review by JosseVanDelm — APPROVED** [review]

**Pull request merged**